### DISCUSSION TITLE
Sunsetting justpy

### DISCUSSION THREAD

**Discussion by WolfgangFahl** [elimintz/justpy#685] [General]
justpy has been a great project and our goal to reach 1000 stars soon was reached now.
Given that nicegui is somewhat of a fork of justpy and is much more successful with >4000 stars there is not much use to try to build on justpy and move it further. 

Therefore I intend to close most issues now that ask for enhancements pointing to nicegui as an alternative that might implement these features and concentrate on helping people migrate to nicegui or keep their existing justpy code running for a while. 

justpy 0.13 seems to be sufficiently stable to not cause too many worries.

**What do you think?**

**Comment by freemant2000** [comment]
What does @elimintz think? Is he also migrating to NiceGUI?

**Reply by richlysakowski** [reply]
I just tried installing and noticed that none of the files have extensions.  jpdemo does not run. 
This package was not written to run on Windows.  It looks like it is hard-coded to run on Mac or Linux only.  I'll pass on this, especially since you passed on maintenance and moved to nicegui (another very "mackie" framework.)  I was interested in this because of its integration of Aggrid and Flask.

**Reply by WolfgangFahl** [reply]
thanks for the feedback

**Reply by freemant2000** [reply]
My student and I have made all the tests run on Windows (and of course Linux). However, we are in the middle of upgrading it to Vue 3 and it is not in a stable state for now.

**Comment by msrkala** [comment]
Yes, one should think pragmatically.

**Comment by WolfgangFahl** [comment]
Just released version 14.0 after checking that the github and jenkins CI still works. Migration guide hints are welcome - just add articles to the discussion if you successfully migrated code from justpy to nicegui.

**Comment by freemant2000** [comment]
If someone needs help fixing existing bugs in justpy, he/she may contact me. I have been teaching a high schooler Python programming and he is largely capable of fixing most bugs under my guidance.

**Comment by freemant2000** [comment]
My student has upgraded the html_component in justpy to Vue 3. You can see the test html file at https://github.com/GavinLeongMO/justpy/blob/master/tests/html_component_test_page.html .

**Reply by sandeep-gh** [reply]
Hi @freemant2000, I thought justpy was already at Vue3. In any case, were there any API level changes when moving to  Vue3 or did we have to simply update the corresponding libraries. 

I published a fork/extension/enhancement to justpy which is using svelte as frontend library. However, I am planning to bring back Vue engine back. We should coordinate, if there are api level changes.  There are inefficiencies in the current Justpy that I have addressed in my repo. It would be good to upgrade to Vue3 there as well.

**Reply by freemant2000** [reply]
Sure. We intend to keep the internal justpy protocol unchanged while upgrading to Vue v3, so the components and existing justpy code should continue to work.

**Comment by WolfgangFahl** [comment]
I made justpy's demo available as components in the nicegui solutions bazaar at http://ngdemo.bitplan.com/components/justpy-org_justpy

So if you need any of these "components" converted to justpy please file an issue in https://github.com/WolfgangFahl/nicegui_widgets/issues/new/choose

**Comment by bitnom** [comment]
Are there any benefits to justpy over nicegui? I read the nicegui readme but it doesn't go into much detail about the differences.

**Reply by falkoschindler** [reply]
In short, NiceGUI
- uses current versions of Vue, Quasar and Tailwind,
- is based on FastAPI and supports all of its features,
- has a larger and more active community, and
- is actively developed and maintained.

The number of UI elements might also be larger, but it's hard to say for sure. The intention was also to create more high-level elements than HTML's basic `div`, `a`, ... Have a look into the [documentation](https://nicegui.io/) where you'll find lots of interactive demos showcasing features.

**Reply by freemant2000** [reply]
Certainly, for example, I am teaching my students web technologies (html, css, js) and Python, so Justpy allows them to work all these. With NiceGUI, HTML is kind of hidden away, so they would be learning just yet another UI framework.

**Reply by WolfgangFahl** [reply]
at http://ngdemo.bitplan.com/solutions you can see how the community is evolving. The components in justpy have not been really components in the past but just examples from the tutorial. A component base approach is much more feasible with nicegui.

**Comment by psy0rz** [comment]
thanks for all your hard work!

**Reply by WolfgangFahl** [reply]
@psy0rz appreciated - open source is a way of life.
